xgboost 0.90 (new formula)

ankane opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

SMillerDev commented:
Please don't specify the amount of jobs for make. That might cause issues on other systems.

ankane commented:
My bad, updated

javian commented on Formula/xgboost.rb:
```diff
@@ -0,0 +1,26 @@
+class Xgboost < Formula
+  desc "Scalable, Portable and Distributed Gradient Boosting Library"
+  homepage "https://xgboost.ai/"
+  url "https://github.com/dmlc/xgboost.git",
```
Is there a reason that the git repo is used as opposed to the tarball which is the preferred option?

ankane replied:
Yeah, the repo has submodules. I created an earlier version with resources (link below), but this seemed cleaner/more maintainable.

https://github.com/Homebrew/homebrew-core/compare/master...ankane:xgboost1

javian commented:
The docs say it requires gcc to enable multi-threading - do you think that would be necessary for this Formula?

ankane commented:
Yeah, that'd be ideal. Just pushed an update.

ankane commented:
It looks like `brew linkage --test xgboost` is failing for each build. However, I'm not sure how to reproduce it locally. Are you familiar with this error?

ankane commented:
Sorry the noise. It was due to having `gcc@8` marked as `build` when it's `libgomp` is needed during runtime. The formula now supports multi-threading.

fxcoudert commented on Formula/xgboost.rb:
```diff
@@ -0,0 +1,30 @@
+      :revision => "515f5f5c4779ff5361dcd796e22d55937e1048ea"
+
+  depends_on "cmake" => :build
+  depends_on "gcc@8"
```
Sorry to disagree with @javian there, but unless multithreading is essential to the app's behaviour, we generally avoid it. GCC is a heavy dependency, so we want to avoid it if at all possible, and compile with the system compiler.

Using `libomp` maybe help provide OpenMP support, if that's what you need.

And in any case, compiling with an older version of GCC is a definite “no way”.

ankane replied:
Hey @fxcoudert, thanks for the feedback on this as well. It looks @hcho3 has it working with `libomp` but it isn't merged yet, so will see what the maintainers want to do: https://github.com/dmlc/xgboost/issues/4477

ankane commented:
Hey @fxcoudert, I updated the formula to work with Apple Clang (disabled OpenMP). Also switched to `make install` (based on your feedback from the `lightgbm` formula). It looks like the build failed due to a credentials error.

SMillerDev commented:
@BrewTestBot test this please

SMillerDev commented:
Thanks @ankane! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

ankane commented:
Thanks @SMillerDev and @fxcoudert 👍
